its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/ScannerMSBuildTest.java

```diff
@@ -1361,27 +1373,17 @@ private void validateCSharpSdk(String folderName) throws IOException {
 
   private void assertUIWarnings(BuildResult buildResult) {
     // AnalysisWarningsSensor was implemented starting from analyzer version 8.39.0.47922 (https://github.com/SonarSource/sonar-dotnet-enterprise/commit/39baabb01799aa1945ac5c80d150f173e6ada45f)
-    var analyzerVersion = TestUtils.getAnalyzerVersion(ORCHESTRATOR);
-    if (!TestUtils.isDevOrLatestRelease(analyzerVersion)
-      && !Version.create(analyzerVersion).isGreaterThan(8, 39)) {
-      return;
-    }
-    var warnings = TestUtils.getAnalysisWarningsTask(ORCHESTRATOR, buildResult);
-    assertThat(warnings.getStatus()).isEqualTo(Ce.TaskStatus.SUCCESS);
-    var warningsList = warnings.getWarningsList();
-    assertThat(warningsList.stream().anyMatch(
-      // The warning is appended to the timestamp, we want to assert only the message
-      x -> x.endsWith("Multi-Language analysis is enabled. If this was not intended and you have issues such as hitting your LOC limit or analyzing unwanted files, please set " +
-        "\"/d:sonar.scanner.scanAll=false\" in the begin step.")
-    )).isTrue();
+    // So it's available from SQ 9.9 onwards
     if (ORCHESTRATOR.getServer().version().isGreaterThanOrEquals(9, 9)) {
-      assertThat(warningsList.size()).isEqualTo(1);
-    } else {
+      var warnings = TestUtils.getAnalysisWarningsTask(ORCHESTRATOR, buildResult);
+      assertThat(warnings.getStatus()).isEqualTo(Ce.TaskStatus.SUCCESS);
+      var warningsList = warnings.getWarningsList();
       assertThat(warningsList.stream().anyMatch(
         // The warning is appended to the timestamp, we want to assert only the message
-        x -> x.endsWith("Starting in January 2025, the SonarScanner for .NET will not support SonarQube versions below 9.9. Please upgrade to a newer version.")
+        x -> x.endsWith("Multi-Language analysis is enabled. If this was not intended and you have issues such as hitting your LOC limit or analyzing unwanted files, please set " +
+          "\"/d:sonar.scanner.scanAll=false\" in the begin step.")
       )).isTrue();
-      assertThat(warningsList.size()).isEqualTo(2);
+      assertThat(warningsList.size()).isEqualTo(1);
```

Comment:
You removed the else branch (testing SQ<9.9)and the assertion of `"Starting in January 2025, the SonarScanner for .NET will not support SonarQube versions below 9.9. Please upgrade to a newer version."`.

Reply:
Yes, that was on purpose. The else was practically unreachable anyway.
The was an early bail out condition at the beginning that was not executing this method for analyzer version below 8.39.

In the past, this condition needed to be based on analyzer version, because we were in the middle of that LTA cycle. So SQ 8.9 LTS, LATEST_RELEASE and DEV had different version of analyzer and needed different behavior. And that was changing also for LATEST_RELEASE once we released 8.39. It was a mess.

Nowadays, the analyzer check for 8.39 is semantically equivalent to SQ check for 9.9, as we're further in the future.

By changing the condition to run the method only for 9.9, it became apparent that the old if/else was never executed in the else branch.